File: Nuclear.ScriptCore/Engine/Components/EntityInfoComponent.cs
using System;
using System.Runtime.CompilerServices;

namespace Nuclear.Components
{

    public class EntityInfoComponent : Component
    {
        //TODO
        //Entity ParentEntity;
        //ECS.Transform Transform;


        public string Name
        {
            get => GetName_Native(Entity.ID);
            set => SetName_Native(Entity.ID, value);
        }

        [MethodImpl(MethodImplOptions.InternalCall)]
        public static extern string GetName_Native(uint entity);

        [MethodImpl(MethodImplOptions.InternalCall)]
        public static extern void SetName_Native(uint entity, string name);

    }

}

File: Nuclear.Engine/Source/Components/EntityInfoComponent.cpp
#include "Components/EntityInfoComponent.h"

namespace Nuclear
{
	namespace Components
	{
		EntityInfoComponent::EntityInfoComponent(const ECS::Transform &Transform, const std::string& name)
			: mName(name), mTransform(Transform)
		{
		}

		EntityInfoComponent::~EntityInfoComponent()
		{
		}
	}
}